sdk/keyvault/keyvault-certificates/.eslintrc.json

```diff
@@ -2,6 +2,19 @@
   "plugins": ["@azure/azure-sdk"],
   "extends": ["../../.eslintrc.json", "plugin:@azure/azure-sdk/recommended"],
   "parserOptions": {
-    "createDefaultProgram": true
-  }
+    "project": "tsconfig.eslint.json"
+  },
+  "ignorePatterns": ["src/core"],
+  "rules": {
+    "@typescript-eslint/no-this-alias": "off",
+    "no-invalid-this": "off"
```

Comment:
I understand why you turn off `no-this-alias`, but where are we hitting `no-invalid-this`?

Reply:
Let me send you the output if I remove this.

Reply:
```
Executing: "eslint package.json src test samples --ext .ts"


/home/vsonline/workspace/sdk/keyvault/keyvault-certificates/src/index.ts
  402:24  error  Unexpected 'this'  no-invalid-this
  402:52  error  Unexpected 'this'  no-invalid-this
  403:35  error  Unexpected 'this'  no-invalid-this
  403:74  error  Unexpected 'this'  no-invalid-this
  404:32  error  Unexpected 'this'  no-invalid-this
  404:68  error  Unexpected 'this'  no-invalid-this
  405:33  error  Unexpected 'this'  no-invalid-this
  405:70  error  Unexpected 'this'  no-invalid-this
  406:21  error  Unexpected 'this'  no-invalid-this
  406:46  error  Unexpected 'this'  no-invalid-this
  407:24  error  Unexpected 'this'  no-invalid-this
  407:52  error  Unexpected 'this'  no-invalid-this
  408:28  error  Unexpected 'this'  no-invalid-this
  408:60  error  Unexpected 'this'  no-invalid-this

/home/vsonline/workspace/sdk/keyvault/keyvault-certificates/src/lro/create/operation.ts
   73:17  error  Unexpected 'this'  no-invalid-this
  120:17  error  Unexpected 'this'  no-invalid-this
  134:8   error  Unexpected 'this'  no-invalid-this
  145:12  error  Unexpected 'this'  no-invalid-this

/home/vsonline/workspace/sdk/keyvault/keyvault-certificates/src/lro/delete/operation.ts
   55:17  error  Unexpected 'this'  no-invalid-this
  105:12  error  Unexpected 'this'  no-invalid-this

/home/vsonline/workspace/sdk/keyvault/keyvault-certificates/src/lro/operation/operation.ts
   73:17  error  Unexpected 'this'  no-invalid-this
  116:17  error  Unexpected 'this'  no-invalid-this
  131:8   error  Unexpected 'this'  no-invalid-this
  142:12  error  Unexpected 'this'  no-invalid-this

/home/vsonline/workspace/sdk/keyvault/keyvault-certificates/src/lro/recover/operation.ts
   58:17  error  Unexpected 'this'  no-invalid-this
  112:12  error  Unexpected 'this'  no-invalid-this

/home/vsonline/workspace/sdk/keyvault/keyvault-certificates/test/CRUD.test.ts
   27:47  error  Unexpected 'this'  no-invalid-this
   41:64  error  Unexpected 'this'  no-invalid-this
   58:64  error  Unexpected 'this'  no-invalid-this
   74:66  error  Unexpected 'this'  no-invalid-this
  108:64  error  Unexpected 'this'  no-invalid-this
  133:66  error  Unexpected 'this'  no-invalid-this
  154:64  error  Unexpected 'this'  no-invalid-this
  172:66  error  Unexpected 'this'  no-invalid-this
  185:64  error  Unexpected 'this'  no-invalid-this
  203:64  error  Unexpected 'this'  no-invalid-this
  219:64  error  Unexpected 'this'  no-invalid-this
  248:66  error  Unexpected 'this'  no-invalid-this
  266:64  error  Unexpected 'this'  no-invalid-this
  282:64  error  Unexpected 'this'  no-invalid-this
  299:64  error  Unexpected 'this'  no-invalid-this
  315:59  error  Unexpected 'this'  no-invalid-this
  316:64  error  Unexpected 'this'  no-invalid-this
  381:64  error  Unexpected 'this'  no-invalid-this
  403:64  error  Unexpected 'this'  no-invalid-this

/home/vsonline/workspace/sdk/keyvault/keyvault-certificates/test/list.test.ts
   28:47  error  Unexpected 'this'  no-invalid-this
   65:64  error  Unexpected 'this'  no-invalid-this
   91:66  error  Unexpected 'this'  no-invalid-this
  117:64  error  Unexpected 'this'  no-invalid-this
  149:64  error  Unexpected 'this'  no-invalid-this
  184:64  error  Unexpected 'this'  no-invalid-this
  242:64  error  Unexpected 'this'  no-invalid-this

/home/vsonline/workspace/sdk/keyvault/keyvault-certificates/test/lro.create.test.ts
  20:47  error  Unexpected 'this'  no-invalid-this
  34:75  error  Unexpected 'this'  no-invalid-this
  56:75  error  Unexpected 'this'  no-invalid-this

/home/vsonline/workspace/sdk/keyvault/keyvault-certificates/test/lro.delete.test.ts
  20:47  error  Unexpected 'this'  no-invalid-this
  34:75  error  Unexpected 'this'  no-invalid-this
  57:75  error  Unexpected 'this'  no-invalid-this

/home/vsonline/workspace/sdk/keyvault/keyvault-certificates/test/lro.operation.test.ts
  19:47  error  Unexpected 'this'  no-invalid-this
  33:75  error  Unexpected 'this'  no-invalid-this
  60:75  error  Unexpected 'this'  no-invalid-this

/home/vsonline/workspace/sdk/keyvault/keyvault-certificates/test/lro.recover.test.ts
   22:47  error  Unexpected 'this'  no-invalid-this
   37:31  error  Unexpected 'this'  no-invalid-this
   69:31  error  Unexpected 'this'  no-invalid-this
  116:33  error  Unexpected 'this'  no-invalid-this

/home/vsonline/workspace/sdk/keyvault/keyvault-certificates/test/mergeAndImport.test.ts
  26:47  error  Unexpected 'this'  no-invalid-this
  43:64  error  Unexpected 'this'  no-invalid-this
  75:66  error  Unexpected 'this'  no-invalid-this

/home/vsonline/workspace/sdk/keyvault/keyvault-certificates/test/recoverBackupRestore.test.ts
   26:47  error  Unexpected 'this'  no-invalid-this
   40:64  error  Unexpected 'this'  no-invalid-this
   67:64  error  Unexpected 'this'  no-invalid-this
   80:64  error  Unexpected 'this'  no-invalid-this
  126:66  error  Unexpected 'this'  no-invalid-this

✖ 73 problems (73 errors, 0 warnings)

The script failed with exit code 1
```

Reply:
This eslint rule gets triggered a lot. For example, we're using `this` in the tests to get the test titles from mocha, so that the resources created can have a meaningful relationship with the tests.

Reply:
In the LRO code, for example, I'm reading an updating a property that exists as part of the plain object of the operation. Operations are not classes, only plain objects, and they have an update method that makes use of other properties at `this`, if that makes sense.

Reply:
IIRC on F12 had a standard "disable these rules" comment at the top of our mocha test files, one of which was this rule, for the same reason. I am not sure if we should disable the rule for the whole project though.

Furthermore, I think we might want to consider excluding tests from linting if there is a high overhead.

Reply:
+1 to disabling linting on tests. I don't think it's super necessary and it seems to have lots of overhead...

Reply:
I'll do that change, but that doesn't address the code related to core-lro.

Reply:
Done, now we're not linting tests. I reverted the lint fixes on the test files too, to make this easier to review.